README.md

```diff
@@ -57,3 +57,65 @@ If you need to declare dependencies that are not ROS packages, typically you can
 a `package.xml`. If it is an unreleased source dependency, (ex. the roboteq c++ driver we wrote), declare it in the 
 `metapackage_dependencies.rosinstall`. Source code modules are still subject to the clangformat and clangtidy checks 
 because they should only consist of code the team has written. 
+
+## Git Workflow
+### Issues
+Before writing code, assign yourself the issue that pertains to the code you are going to write. If there is no GitHub issue created yet, create one for yourself! Try to keep all discussions related to the task within the issue.
+
+### Branch Naming
+To be able to quickly see work done by members, you should name you're branches with the following scheme:
+`user/<username>/#<issue number>/<issue name>`
+
+For example:
+`user/wmmc88/#2/update-readme-with-git-workflow`
+
+### Pull Requests
+To merge code into master, you must open a pull request. Branches must be up to date with the latest master to be able to merge. Choose people you think are familiar with your task to do code reviews of your code. 
+After your PR is merged, make sure you close the associated issue and delete the branch. Github will usually delete the branch automatically and you can automatically close the issue by mentioning `closes #<issue number>` in the PR or in one of the commits on the branch.
+
+If you have code that's not ready to merge, but you'd still like people thoughts on it, open a [draft pull request](https://github.blog/2019-02-14-introducing-draft-pull-requests/).
+
+## Running CI Stages Locally
```

Comment:
does this mean i don't have to spam amended commits?

Reply:
yea. you can pretty much simulate all the stages except for arm64 build, which is honestly not worth setting up locally to run